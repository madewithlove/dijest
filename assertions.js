module.exports = require("./dist/assertions");
