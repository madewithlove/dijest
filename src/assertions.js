export {default as component} from './assertions/component';
export {default as reducer} from './assertions/reducer';
export {default as saga} from './assertions/saga';
