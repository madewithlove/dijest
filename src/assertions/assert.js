export default (input, expected) => {
    // If we specified a result to test against
    // use that instead of a snapshot
    if (typeof expected !== "undefined") {
        expect(input).toEqual(expected);

        return;
    }

    expect(input).toMatchSnapshot();
};
