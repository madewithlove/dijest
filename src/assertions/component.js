import renderer from 'react-test-renderer';

/**
 * Assert that a component equals X
 */
export default component => {
    if (typeof component.toJSON === 'undefined') {
        component = renderer.create(component);
    }

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    return {component, tree};
}