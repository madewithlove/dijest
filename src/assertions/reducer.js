import {fromJS} from 'immutable';

/**
 * Assert that a reducer equals X
 */
export default (actual, before, action = {}, after) => {
    before = fromJS(before);
    const result = actual(before, action);

    // If we specified a result to test against
    // use that instead of a snapshot
    if (after) {
        return expect(result.toJS()).toEqual(after);
    }

    return expect(result).toMatchSnapshot();
};
