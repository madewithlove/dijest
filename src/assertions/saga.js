/**
 * Assert that a saga equals X
 */
export default (actual, effects) => {
    let result;

    // If we provided specific results to test against
    // use those instead of a snapshot
    if (typeof effects !== 'undefined') {
        let next = actual.next();

        return effects.forEach(effect => {
            expect(next.value).toEqual(effect.action);
            next = actual.next(effect.response);
        });
    }

    // Else test against a snapshot
    const actions = [];
    do {
        result = actual.next();
        if (!result.done) {
            actions.push(result);
        }
    } while (!result.done);

    expect(actions).toMatchSnapshot();
};
