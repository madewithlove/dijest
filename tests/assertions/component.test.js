import React, {Component} from 'react';
import {component as assert} from '../../dist/assertions';

class RealComponent extends Component {
    state = {text: 'foo'};

    onClick() {
        this.setState({text: 'bar'})
    }

    render() {
        return <p onClick={() => this.onClick()}>{this.state.text}</p>;
    }
}

test('can assert stuff about component', () => {
    const {tree, component} = assert(<RealComponent />);

    tree.props.onClick();
    assert(component);
});
