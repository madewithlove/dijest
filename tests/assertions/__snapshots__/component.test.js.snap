exports[`test can assert stuff about component 1`] = `
<p
  onClick={[Function onClick]}>
  foo
</p>
`;

exports[`test can assert stuff about component 2`] = `
<p
  onClick={[Function onClick]}>
  bar
</p>
`;
